nova_vm/src/ecmascript/builtins/numbers_and_dates/math_object.rs

```diff
@@ -342,122 +341,665 @@ impl MathObject {
+        if n.is_neg_infinity(agent) {
+            return Ok(Value::zero());
+        }
+
+        //5. Return an implementation-approximated Number value representing the result of the exponential function of ℝ(n).
+        Ok(Value::from_f64(agent, n.into_f64(agent).exp()))
     }
 
     fn expm1(agent: &mut Agent, _this_value: Value, arguments: ArgumentsList) -> JsResult<Value> {
-        let _x = to_number(agent, arguments.get(0))?;
-        todo!();
+        // 1. Let n be ? ToNumber(x).
+        let n = to_number(agent, arguments.get(0))?;
+
+        // 2. If n is one of NaN, +0𝔽, -0𝔽, or +∞𝔽, return n.
+        if n.is_nan(agent)
+            || n.is_pos_zero(agent)
+            || n.is_neg_zero(agent)
+            || n.is_pos_infinity(agent)
+        {
+            return Ok(n.into_value());
+        }
+
+        // 3. If n is -∞𝔽, return -1𝔽.
+        if n.is_neg_infinity(agent) {
+            return Ok(Value::from(-1));
+        }
+
+        // 4. Return an implementation-approximated Number value representing the result of subtracting 1 from the exponential function of ℝ(n).
+        Ok(Value::from_f64(agent, n.into_f64(agent).exp_m1()))
     }
 
     fn floor(agent: &mut Agent, _this_value: Value, arguments: ArgumentsList) -> JsResult<Value> {
-        let _x = to_number(agent, arguments.get(0))?;
-        todo!();
+        // 1. Let n be ? ToNumber(x).
+        let n = to_number(agent, arguments.get(0))?;
+
+        // 4. If n is an integral Number, return n.
+        if let Number::Integer(_) = n {
+            return Ok(n.into_value());
+        }
+
+        // 2. If n is not finite or n is either +0𝔽 or -0𝔽, return n.
+        if !n.is_finite(agent) || n.is_pos_zero(agent) || n.is_neg_zero(agent) {
+            return Ok(n.into_value());
+        }
+
+        let n = n.into_f64(agent);
+
+        // 3. If n < 1𝔽 and n > +0𝔽, return +0𝔽.
+        if n < 1.0 && n > 0.0 {
+            return Ok(Value::zero());
+        }
+
+        // 5. Return the greatest (closest to +∞) integral Number value that is not greater than n.
+        Ok(Value::from_f64(agent, n.floor()))
     }
 
     fn fround(agent: &mut Agent, _this_value: Value, arguments: ArgumentsList) -> JsResult<Value> {
-        let _x = to_number(agent, arguments.get(0))?;
-        todo!();
+        // 1. Let n be ? ToNumber(x).
+        let n = to_number(agent, arguments.get(0))?;
+
+        // 2. If n is NaN, return NaN.
+        if n.is_nan(agent) {
+            return Ok(Value::nan());
+        }
+
+        // 3. If n is one of +0𝔽, -0𝔽, +∞𝔽, or -∞𝔽, return n.
+        if n.is_pos_zero(agent)
+            || n.is_neg_zero(agent)
+            || n.is_pos_infinity(agent)
+            || n.is_neg_infinity(agent)
+        {
+            return Ok(n.into_value());
+        }
+
+        // 4. Let n32 be the result of converting n to IEEE 754-2019 binary32 format using roundTiesToEven mode.
+        let n32 = n.into_f32(agent).round_ties_even();
```

Comment:
This is wrong, `f32::round_ties_even()` rounds the number *to an integer*, using the round ties to even rounding mode. The spec asks to round ties to even *in the conversion to `f32`*, and in fact, that rounding mode is the one that Rust already uses in `n as f32`: https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast